makefile

```diff
@@ -0,0 +1,37 @@
+SOURCE_DIR  := ./src
+INCLUDE_DIR := ./include
+BUILD_DIR   := ./build
+
+CC        := gcc
+CFLAGS    := -Wall
+LDFLAGS   := -lm
+INC_FLAGS := -I$(INCLUDE_DIR)
+
+DEPS = $(INCLUDE_DIR)/creator.h $(INCLUDE_DIR)/finalizer.h $(INCLUDE_DIR)/consumer.h $(INCLUDE_DIR)/producer.h $(INCLUDE_DIR)/buffer.h
+
+.PHONY: all
+all: $(BUILD_DIR) $(BUILD_DIR)/creator $(BUILD_DIR)/producer $(BUILD_DIR)/consumer $(BUILD_DIR)/finalizer
+
+$(BUILD_DIR)/%.o: $(SOURCE_DIR)/%.c $(DEPS)
+	$(CC) -c -o $@ $< $(CFLAGS)
```

Comment:
Please add "$(INC_FLAGS)" to this rule, this is necessary because we are using an include dir. If we had sources and headers in the same dir, it wouldn't be required.